src/stream-tcp.c

```diff
@@ -430,13 +430,10 @@ void StreamTcpInitConfig(char quiet)
         /* checking for "auto" and falling back to boolean to provide
          * backward compatibility */
         if (strcmp(temp_stream_inline_str, "auto") == 0) {
-            if (EngineModeIsIPS()) {
-                stream_inline = 1;
-            } else {
-                stream_inline = 0;
-            }
+            stream_inline = AUTO;
         } else if (ConfGetBool("stream.inline", &inl) == 1) {
-            stream_inline = inl;
+            if (inl == IPS)
```

Comment:
This doesn't look right. ConfGetBool will not set IPS to inl. Only by chance (IPS == 1). Also, if stream.inline is true, it should set IPS to stream_inline unconditionally.